coordinator/meshapi.go

```diff
@@ -108,7 +110,7 @@ func (i *meshAPIServer) NewMeshCert(ctx context.Context, _ *meshapi.NewMeshCertR
 	hostData := manifest.NewHexString(report.HostData())
 	entry, ok := state.Manifest.Policies[hostData]
 	if !ok {
-		return nil, fmt.Errorf("report data %s not found in manifest", hostData)
+		return nil, status.Errorf(codes.PermissionDenied, "policy hash %s not found in manifest", hostData)
```

Comment:
I'm not too happy about `PermissionDenied`. To me, `PermissionDenied` primarily indicates an authentication problem and not a problem related to unexpected inputs. To quote the official docs:

> It must not be used if the caller cannot be identified (use Unauthenticated instead for those errors).

Personally I'd prefer either `InvalidArgument` or `NotFound`.

Reply:
The part you quoted does not apply, though: we don't have an _authentication_ error here, because we know the caller's policy hash, which is the identity for Contrast.

`InvalidArgument` is not applicable because it "indicates arguments that are problematic regardless of the state of the system". The suggested alternative `FailedPrecondition` would be an option, but imho `PermissionDenied` is more precise: "you don't have permission to get a certificate for this Contrast generation". How about I change the error string to make this more clear?

Reply:
Sorry, I meant to also say in the first paragraph that `PermissionDenied` is an _authorization_ error and `Unauthenticated` is the *authentication* error. Maybe it makes more sense that way?

Reply:
Sure, let's keep using `PermissionDenied`.